datumaro/components/annotation.py

```diff
@@ -143,23 +143,23 @@ def __attrs_post_init__(self):
     def _reindex(self):
         indices = {}
         for index, item in enumerate(self.items):
-            assert item.name not in self._indices
-            indices[item.name] = index
+            assert (item.parent + item.name) not in self._indices
+            indices[item.parent + item.name] = index
         self._indices = indices
 
     def add(
         self, name: str, parent: Optional[str] = None, attributes: Optional[Set[str]] = None
     ) -> int:
         assert name
-        assert name not in self._indices, name
+        assert (parent if parent else "") + name not in self._indices, name
```

Comment:
Message in this assert is not fully correct, suggestion:
```suggestion
        assert (parent if parent else "") + name not in self._indices, (parent if parent else "") + name
```
or without message at all
```suggestion
        assert (parent if parent else "") + name not in self._indices
```
or any other assert with readable message